aviatrix/resource_aviatrix_spoke_external_device_conn.go

```diff
@@ -1113,6 +1235,74 @@ func resourceAviatrixSpokeExternalDeviceConnUpdate(d *schema.ResourceData, meta
 		}
 	}
 
+	enableBfd, ok := d.Get("enable_bfd").(bool)
+	if !ok {
+		return fmt.Errorf("expected enable_bfd to be a boolean, but got %T", d.Get("enable_bfd"))
+	}
+	if enableBfd && connType != "bgp" {
+		return fmt.Errorf("BFD is only supported for BGP connection type")
+	}
+	if enableBfd {
+		// get the new BGP BFD config
+		bgpBfdConfig, ok := d.Get("bgp_bfd").([]interface{})
+		if !ok {
+			return fmt.Errorf("expected bgp_bfd to be a list of maps, but got %T", d.Get("bgp_bfd"))
+		}
+		var bgpBfdConfigList []*goaviatrix.BgpBfdConfig
+		// Update the BGP BFD config if bfd is enabled and config has changed
+		if len(bgpBfdConfig) > 0 && d.HasChange("bgp_bfd") {
```

Comment:
I'm not sure I follow this. If you have the config, I think you should use the values from that whether or not HasChange is true. If HasChange is false, is forcing the defaults the right things to do?

Reply:
Updated the logic to follow this:
if bfd is enabled and bgp_bfd config is changed then set the new config
else if bfd is enabled but no bgp_bfd config is provided then use the default
else if bfd is disabled then remove the bgp_bfd config
else no change.